src/CBLURLEndpointListener.cc

```diff
@@ -0,0 +1,216 @@
+//
+//  CBLURLEndpointListener.cc
+//
+// Copyright © 2025 Couchbase. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "CBLURLEndpointListener_Internal.hh"
+
+#ifdef COUCHBASE_ENTERPRISE
+
+using namespace fleece;
+
+std::mutex CBLURLEndpointListener::_mutex;
+constexpr CBLTimestamp kCBLAnonymousIdentityMinValidTimeAllowed = 86400; /* 24 Hours */
+
+bool CBLURLEndpointListener::start() {
```

Comment:
I think this method can be void instead of returning boolean. When the listener cannot be started, it could throw an error instead so that C level can convert that to an out error.

Reply:
okay will do